### PR TITLE
Transport 2.0

### PR DESCRIPTION
**Pull request by charlyqchm**
The next modifications have been made:

- Transport now can work with more than 2 electrodes. The number
  of electrodes must be declared in the input with the variable
  nbias.
- Now the electrodes can be activated in a specific moment of TD.
  The activation step must be declared in the transport.dat file.
- The info of the file atomgroup now must be written it in transport.dat.
- Van Voorhis potentials (fockbias_subs) has been updated to work
  with transport and Close and Open Shell. The potential for each
  atom must be declared in a file called atombias.in.

### CODE REVIEW

**Comment by ramirezfranciscof** [comment]
@charlyqchm This is a check failure notifications.

**Comment by ramirezfranciscof** [comment]
> The activation step must be declared in the transport.dat file.
> The info of the file atomgroup now must be written it in transport.dat.

It is not clear to me from these two statements whether "transport.dat" is an input or output file. If its an input, can we name it "transport.in"? I feel like we should leave ".dat" extensions for outputs.

File: lioamber/SCF.f90 [hunk 1]
```diff
@@ -764,10 +768,13 @@ subroutine SCF(E)
         call rho_bop%Gets_data_AO(rho_b)
         call messup_densmat( rho_b )
 
-!carlos: Beta Energy storage in RMM
-        do kk=1,M
-          RMM(M22+kk-1) = morb_energy(kk)
-        end do
+!carlos: Beta Energy stored
+         Eorbs_b=morb_energy
+!charly: The storage in RMM it seems that is not necessary anymore
```
Comment (fedepedron) [inline lioamber/SCF.f90]: Remove the commented code.

File: lioamber/cublasmath/cupredictor.f [hunk 2]
```diff
@@ -140,6 +144,10 @@ subroutine cupredictor_DZ(F1a,F1b,FON,rho2,devPtrX,factorial,
 
        if (OPEN) then
           call spunpack('L',M,RMM(M3),FBA(MTB+1:MTB+M,MTB+1:MTB+M,2))
+
+!Fockbias:
```
Comment (fedepedron) [inline lioamber/cublasmath/cupredictor.f]: Remove these redundant comments.

File: lioamber/SCF.f90 (hunk 1 above)
Comment (fedepedron) [inline lioamber/SCF.f90]: Please remove the commented code.

File: lioamber/TD.f90 [hunk 3]
```diff
@@ -1026,6 +1028,10 @@ subroutine td_bc_fock_cu(M_in,M, MM, RMM5, fock_op, devPtrX, natom, nshell,    &
    else
       fock=fock_0
    end if
+
+!Fockbias is applied
```
Comment (fedepedron) [inline lioamber/TD.f90]: Remove all "!fockbias" redundant comments, it is obvious from the subroutine name.

File: lioamber/fockbias_subs.f90 [hunk 4]
```diff
@@ -34,25 +34,40 @@ module fockbias_subs
...
-
-   fockbias_is_active = activate
-   fockbias_is_shaped = do_shape
-   fockbias_timegrow  = grow_start
-   fockbias_timefall  = fall_start
-   fockbias_timeamp0  = setamp
-
-end subroutine fockbias_setup0
+!carlos:
+!fockbias_setup0 is not neccesary for the moment because the values of the
```
Comment (fedepedron) [inline lioamber/fockbias_subs.f90]: Does this still serve any purpose? If not, remove this whole section.

File: lioamber/fockbias_subs.f90 [hunk 5]
```diff
@@ -143,7 +158,10 @@ subroutine fockbias_loads( Natom, atom_of_orb, file_unit_in, file_name_in )
    end if
 
 !  If there is input filename, save it so as to know the source.
-   if ( present(file_name_in) ) fockbias_readfile = file_name_in
+
+!charly: comente esto temporalmente
+!   if ( present(file_name_in) ) fockbias_readfile = file_name_in
+
```
Comment (fedepedron) [inline lioamber/fockbias_subs.f90]: until when?

File: lioamber/fockbias_subs.f90 [hunk 6]
```diff
@@ -241,6 +259,10 @@ subroutine fockbias_apply_d( timepos, fockmat )
 
    end if
 
+   if (time_shape < 1.00d-16) time_shape = 0.0d0
+
+   print*, "bias", time_shape
+
```
Comment (fedepedron) [inline lioamber/fockbias_subs.f90]: Please standardize the output or remove  this print.

File: lioamber/transport.f90 [hunk 7]
```diff
@@ -41,13 +44,13 @@ subroutine transport_init(M, dim3, natom, Nuc, RMM5, overlap, rho, OPEN)
...
 
-   ngroup  = 0
-      allocate(rhofirst(M,M,dim3))
+   allocate(rhofirst(M,M,dim3),timestep_init(nbias))
 
    allocate(group(natom), mapmat(M,M))
    call transport_read_groups(natom)
    call mat_map(Nuc, M, natom)
 
+!charly: is this necessary?
```
Comment (fedepedron) [inline lioamber/transport.f90]: if YOU of all people do not know, no one knows XD

**Pull request merged**